html/includes/functions.inc.php

```diff
@@ -704,7 +704,62 @@ function generate_port_link($port, $text = null, $type = null, $overlib = 1, $si
     }
 }//end generate_port_link()
 
+function generate_sensor_link($args, $text = null, $type = null)
+{
+    global $config;
+
+    $args = cleanPort($args);
+
+    if (!$text) {
+        $text = fixIfName($args['sensor_descr']);
+    }
+
+    if (!$type) {
+        $args['graph_type'] = "sensor_" . $args['sensor_class'];
+    } else {
+        $args['graph_type'] = "sensor_" . $type;
+    }
+
+    if (!isset($args['hostname'])) {
+        $args = array_merge($args, device_by_id_cache($args['device_id']));
+    }
+
+    $content = '<div class=list-large>' . $text . '</div>';
 
+    $content .= "<div style=\'width: 850px\'>";
+    $graph_array = array();
+    $graph_array['type'] = $args['graph_type'];
+    $graph_array['legend'] = 'yes';
+    $graph_array['height'] = '100';
+    $graph_array['width'] = '340';
+    $graph_array['to'] = $config['time']['now'];
+    $graph_array['from'] = $config['time']['day'];
+    $graph_array['id'] = $args['sensor_id'];
+    $content .= generate_graph_tag($graph_array);
+    $graph_array['from'] = $config['time']['week'];
+    $content .= generate_graph_tag($graph_array);
+    $graph_array['from'] = $config['time']['month'];
+    $content .= generate_graph_tag($graph_array);
+    $graph_array['from'] = $config['time']['year'];
+    $content .= generate_graph_tag($graph_array);
+    $content .= '</div>';
+
+
+    $url = generate_sensor_url($args);
+    if (port_permitted($args['interface_id'], $args['device_id'])) {
+        return overlib_link($url, $text, $content, null);
+    } else {
+        return fixifName($text);
+    }
+}//end generate_sensor_link()
+
+
+function generate_sensor_url($sensor, $vars = array())
+{
+    return generate_url(array('page' => 'graphs', 'id' => $sensor['sensor_id'], 'type' => $sensor['graph_type'], 'from' => $config['time']['day']), $vars);
```

Comment:
$config was not pulled from global here.  Instead of doing that, use the \LibreNMS\Config class.  We are slowly removing $config from the code.

Reply:
done